src/aleph/http/client.clj

```diff
@@ -147,11 +147,11 @@
 
 (defn client-handler
   [response-stream ^long buffer-capacity]
-  (let [response (atom nil)
-        buffer (atom [])
+  (let [response (volatile! nil)
+        buffer (volatile! [])
         buffer-size (AtomicInteger. 0)
-        stream (atom nil)
-        complete (atom nil)
+        stream (volatile! nil)
+        complete (volatile! nil)
```

Comment:
Correct me if I'm wrong but on this line https://github.com/ztellman/aleph/blob/master/src/aleph/http.clj#L334-L348 the HTTP client relies on `complete` being an atom so that it subscribe for the event of it becoming true and releasing the connection to the pool. You can't do that with volatiles, obviously.

Reply:
Well, actually it relies on anything that's "deferrable" (and `clojure.lang.IDeref` should work fine). You're right about access from different threads... which might not hurt in this particular case as we're only reading here. But overall it's a slippery slope I don't really want to play on, at least right now. Using non-coercible chains there also feels a bit suspicious, I need to dig more. Regarding this specific PR, I would close it to choose a better time for such changes in future.

Reply:
Oops, sorry, there are name clashes in the code. That `complete` that eventually becomes `:aleph/complete` is a deferred, not an atom. Sorry for the confusion.

Reply:
Oh, no problems. I still think that it's better to bring such changes when better tests suites will be ready to go, so we can be sure that nothing is broken in a non-obvious way.